cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/BaseStorage.java

```diff
@@ -41,4 +43,37 @@ public Map<String, String> getProperties() {
         .map(HashMap::new)
         .orElseGet(HashMap::new);
   }
+
+  /**
+   * Allocates Table Space for the storage.
+   *
+   * <p>Default tableLocation looks like: {endpoint}/{rootPrefix}/{databaseId}/{tableId}-{tableUUID}
+   *
+   * @return the table location where the table data should be stored
+   */
+  @Override
+  public String allocateTableSpace(
+      String databaseId,
+      String tableId,
+      String tableUUID,
+      String tableCreator,
+      boolean skipProvisioning) {
+    Preconditions.checkArgument(databaseId != null, "Database ID cannot be null");
+    Preconditions.checkArgument(tableId != null, "Table ID cannot be null");
+    Preconditions.checkArgument(tableUUID != null, "Table UUID cannot be null");
+    Preconditions.checkState(
+        storageProperties.getTypes().containsKey(getType().getValue()),
+        "Storage properties doesn't contain type: " + getType().getValue());
+    return URI.create(
```

Comment:
Thanks Chris! would like to understand this more, is this accurate impl of your comment?
```
TableLocation allocateStorage(...){
        return getTableLocation();
}
TableLocation getTableLocation(...)
```

----------------------
If so, there were couple of reasons I chose the current design.

- You are correct about potential of TableLocation being a common place for validation logic. But besides that, I'm unable to see the value it might bring in prod code, ie. There won't be any place where we do `tableLocation.getXYZ`. The end state is always going to be `tableLocation.toString` because of iceberg layer. If the missing piece is about validations , its up to each storage to validate if the URL is correct, which can be done in allocateStorageSpace itself.
- it's possible to split into two methods: `allocateStorage`, `getTableLocation`. The reason I chose a single method was to avoid the split and centralize the whole logic into a single method. This way we avoid `getTableLocation` as a utility method and the usage of `allocateStorage` is limited during tableCreation time (either CT or CTAS). Let me know if you feel strongly and we can discuss more.

Reply:
Addressed offline, main concern was about centralizing logic for concatenating strings and not let that logic proliferate. 
The centralize logic exists today and that's in "BaseStorage::AllocateStorage()". The only reason we override this in HdfsStorage, LocalStorage was because of https://github.com/linkedin/openhouse/issues/121

As for `TableLocation`  class as an abstraction, we decided to not proceed right now because String returned by allocateTableStorage is treated as immutable and would only be handed off to  iceberg as string.